R/format_PFN.R

```diff
@@ -0,0 +1,589 @@
+#'date of hatching (DH). These are given as integer number, and represent days after a set
+#'starting date (day 0). Day 0 is assumed to be 31. March every year (to be confirmed).
+#'
+#'\strong{ChickAge}: For every capture, we estimate the age of a chick as the difference between the hatch date
+#'taken from BroodIDFledged (in Individual_data) and the CaptureDate.
+#'
+#'
+#'@inheritParams pipeline_params
+#'
+#' @return Generates either 4 .csv files or 4 data frames in the standard format.
+#' @export
+#'
+
+
+format_PFN <- function(db,
+                       species = NULL,
+                       pop = NULL,
+                       path = ".",
+                       output_type = "R"){
+
+  #Force user to select directory
+  force(db)
+
+  #Determine species codes for filtering
+  if(is.null(species)){
+
+    species <- Species_codes$Code
+
+  }
+
+  #Record start time to estimate processing time.
+  start_time <- Sys.time()
+
+  #Load primary brood data
+  Primary_data <- utils::read.csv(file = paste0(db, "/PFN_PrimaryData_EDartmoor.csv"), na.strings = c("", "?"),colClasses = "character")
+
+  #Load complete PiedFlyNet ringing database
+  CMR_data <- utils::read.csv(file = paste0(db, "/PFN_PrimaryData_All_CMR.csv"), na.strings = c("", "?", "UNK", "-"),colClasses = "character") # TODO:Confirm with Malcolm that SITE = "UNK" refers to unknown/unassigned locations
+
+
+  # PREPARATION: RE-RINGING DATA
+
+  # Find all birds that were re-ringed at least once
+  ReRing <- CMR_data[,c('RING', 'RING2')] %>%
+    dplyr::filter(!is.na(.data$RING2))
+
+  # Collate all ring numbers used for each individual (with max. number of re-ringing events = 3)
+  ReRingTable <- make_ReRingTable(raw_data = ReRing, N = 3) %>%
+
+    # Make new individual ID for re-ringed birds
+    dplyr::mutate(ReRingID = paste0('MULTIRING',.data$ReRingNr)) %>%
+    dplyr::select(.data$RingNr, .data$ReRingID)
+
+
+  # CAPTURE DATA
+
+  message("Compiling capture data....")
+
+  Capture_data <- create_capture_EDM(CMR_data = CMR_data, Primary_data = Primary_data, ReRingTable = ReRingTable)
+
+
+  # BROOD DATA
+
+  message("Compiling brood data...")
+
+  Brood_data <- create_brood_EDM(Primary_data = Primary_data, ReRingTable = ReRingTable)
+
+
+  # INDIVIDUAL DATA
+
+  message("Compiling individual data...")
+
+  Individual_data <- create_individual_EDM(Capture_data = Capture_data)
+
+  # LOCATION DATA
+
+  message("Compiling location data...")
+
+  Location_data <- create_location_EDM(Brood_data = Brood_data, Capture_data = Capture_data)
+
+
+  # WRANGLE DATA FOR EXPORT
+
+  # Capture data: Merge in information on ChickAge (from Brood_data) & remove unnecessary columns
+  ChickAge_data <- Brood_data[,c('BroodID', 'ChickAge')]
+  Capture_data <- dplyr::left_join(Capture_data, ChickAge_data, by = 'BroodID')
+  Capture_data$ChickAge[which(Capture_data$Age_observed > 1)] <- NA_integer_
+
+  Capture_data <- Capture_data %>%
+    dplyr::select(-.data$BroodID)
```

Comment:
Minor, but we want the column order to be ChickAge, ExperimentID (it's currently the other way around)

Reply:
Fixed!